Crawler pod memory padding + auto scaling

ikreymer opened the pull request:
- set memory limit to 1.2x memory request to provide extra padding and avoid OOM
- attempt to resize crawler pods by 1.2 when exceeding 90% of available memory
- do a 'soft OOM' (send extra SIGTERM) to pod when reaching 100% of requested memory, resulting in faster graceful restart, but avoiding a system-instant OOM Kill
- Fixes #1632

ikreymer committed:
memory limit padding + auto-scale crawler pods:
- set memory limit to 1.2x memory request to provide extra padding and avoid OOM
- attempt to resize crawler pods by 1.2 when exceeding 90% of available memory
- do a 'soft OOM' (send extra SIGTERM) to pod when reaching 100% of requested memory, resulting in faster graceful restart

tw4l committed:
Remove page debug logging cluttering logs

tw4l reviewed (approved):
Tested locally and works well! Both the resizing and OOM handling are big improvements.

I also pushed a commit to remove some of the page data debug logging leftover from development that was cluttering the operator logs.

ikreymer committed:
add check for has_pod_metrics

ikreymer committed:
signal sending fixes:
- add exec permission to send signal
- add signalAtMem to avoid resending SIGTERM multiple times
